syntax = "proto3";

package drone;

// üîΩ –£–ö–ê–ñ–ò go_package: –ø—É—Ç—å –¥–æ–ª–∂–µ–Ω —Å–æ–æ—Ç–≤–µ—Ç—Å—Ç–≤–æ–≤–∞—Ç—å —Ç–≤–æ–µ–º—É –º–æ–¥—É–ª—é Go
option go_package = "backend/dronepb;dronepb";

service DroneService {
  rpc SendCoordinates(CoordinatesBatchRequest) returns (CoordinatesResponse);
}

message Coordinate {
  double latitude = 1;
  double longitude = 2;
  double altitude = 3;
  int32 point_order = 4;
}

message CoordinatesBatchRequest {
  int32 drone_id = 1;
  repeated Coordinate coordinates = 2;
}

message CoordinatesResponse {
  string status = 1;
}
